src/components/sidebar/SafeList/index.tsx

```diff
@@ -75,27 +82,69 @@ const SafeList = ({ closeDrawer }: { closeDrawer?: () => void }): ReactElement =
   const hasNoSafes = Object.keys(ownedSafes).length === 0 && Object.keys(addedSafes).length === 0
   const isWelcomePage = router.pathname === AppRoutes.welcome
 
+  const handleFetchQueued = useCallback(async () => {
+    // do not run the function if safeQueuedTxs already populated
+    if (safeQueuedTxs !== undefined) return
+
+    const fetchedQueuedTxs: Record<string, string | undefined> = {}
+
+    for (let [chainId, safes] of Object.entries(addedSafes)) {
+      for (let safeAddress of Object.keys(safes)) {
+        // do not request queued txs again for the current Safe
+        if (currentChainId === chainId && sameAddress(currentSafeAddress, safeAddress) && page) {
+          if (page.results.length === 0) continue
+
+          fetchedQueuedTxs[currentSafeAddress] = `${page.results.filter(isTransactionListItem).length}${
+            page.next ? '+' : ''
+          }`
+          continue
+        }
+
+        try {
+          const result = await getTransactionQueue(chainId, safeAddress)
+
+          if (result.results.length === 0) continue
+
+          fetchedQueuedTxs[safeAddress] = `${result.results.filter(isTransactionListItem).length}${
+            result.next ? '+' : ''
+          }`
```

Comment:
Can we avoid the duplicate code here?

Reply:
Yes. Created a small function to avoid the duplication.